### ISSUE TITLE
`leiningen.cljsbuild.config/compiler-global-dirs` leaks invalid `:externs` and `:libs`

### ISSUE THREAD

**Issue by swannodette** [emezeske/lein-cljsbuild#383]
Trying to solve http://dev.clojure.org/jira/browse/CLJS-994 revealed this issue.

**Comment by cemerick** [comment]
Yeah, that's a vestige of #95, which coped with the lack of a library distribution story back in the day. It'll take a bit of work to excise the cruft.

**Comment by cemerick** [comment]
I hoped a quick fix would work (just removing the offending default configuration), but doing so breaks a ton of tests (written using midje, which I'm loathe to touch).

**Comment by cemerick** [comment]
Looked at this again, but just couldn't bring myself to nuke the relevant (useless) tests from orbit that would allow the elimination of the `closure-js` hack. Not a good idea to do so on the way out, IMO. :-(
